packages/app/src/components/DimensionsPanel/Dialogs/AddToLayoutButton/AddToLayoutButton.js

```diff
@@ -36,31 +40,27 @@ export class AddToLayoutButton extends Component {
         this.props.onClick()
     }
 
-    getAxisMeta = axisIdArray =>
-        axisIdArray.map(axisId =>
-            ADD_TO_LAYOUT_OPTIONS.find(
-                axisMetaObj => axisMetaObj.axisId === axisId
-            )
-        )
-
     renderMenuItems = () =>
-        this.getAxisMeta(getAvailableAxes(this.props.visType))
+        getAvailableAxes(this.props.visType)
             .slice(1)
-            .map(axisMetaObj => (
+            .map(axis => (
                 <MenuItem
                     className={this.props.classes.menuItem}
                     component="li"
-                    key={axisMetaObj.axisId}
-                    onClick={() => this.onUpdate(axisMetaObj.axisId)}
+                    key={axis}
+                    onClick={() => this.onUpdate(axis)}
                 >
-                    {axisMetaObj.name}
+                    {i18n.t(`Add to {{axisName}}`, {
+                        axisName: getAxisNameByLayoutType(
+                            axis,
+                            getLayoutTypeByVisType(this.props.visType)
+                        ),
+                    })}
                 </MenuItem>
             ))
 
     render() {
-        const availableAxisMeta = this.getAxisMeta(
-            getAvailableAxes(this.props.visType)
-        )
+        const availableAxis = getAvailableAxes(this.props.visType)
```

Comment:
Should be `const availableAxes` ?

Reply:
As we're also doing `availableAxis.length`

Reply:
Indeed, fixed